examples/matter-controller/BUILD.gn

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+
+import("${build_root}/config/android_abi.gni")
+import("${chip_root}/build/chip/java/rules.gni")
+import("${chip_root}/build/chip/tools.gni")
+
+android_binary("matter-controller") {
+  output_name = "MatterController.jar"
+
+  deps = [
+    ":android",
```

Comment:
@yufengwangca you need to add controller and setup_payload as dependancy, similar to what i did in https://github.com/project-chip/connectedhomeip/pull/23055/files
  deps = [
    ":android",
    "${chip_root}/src/controller/java",
    "${chip_root}/src/setup_payload/java",
    "${chip_root}/third_party/android_deps:annotation",
  ]

otherwies, device controller java module cannot be imported in java code, like import chip.devicecontroller.ChipDeviceController;
import chip.devicecontroller.ControllerParams;